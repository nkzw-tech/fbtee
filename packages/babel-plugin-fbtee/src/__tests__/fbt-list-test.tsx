import { expect, test } from '@jest/globals';
import {
  jsCodeFbtCallSerializer,
  snapshotTransform,
  withFbtImportStatement,
} from './FbtTestUtil.tsx';

expect.addSnapshotSerializer(jsCodeFbtCallSerializer);

const transform = (input: string) =>
  snapshotTransform(withFbtImportStatement(input));

test('fbt.list()', () => {
  expect(
    transform(
      `fbt(
        'Available Locations: ' + fbt.list('locations', ['Tokyo', 'London', 'Vienna'], null, 'or'),
        'Lists',
      )`,
    ),
  ).toMatchSnapshot();
});

test('<fbt:list>', () => {
  expect(
    transform(
      `const x = (
          <fbt desc="Lists">
            Available Locations: <fbt:list name="locations" items={["Tokyo", "London", "Vienna"]} />.
          </fbt>
        );`,
    ),
  ).toMatchSnapshot();

  expect(
    transform(
      `const x = (
          <fbt desc="Lists">
            Available Locations: <fbt:list name="locations" conjunction="and" items={["Tokyo", "London", "Vienna"]} />.
          </fbt>
        );`,
    ),
  ).toMatchSnapshot();

  expect(
    transform(
      `const x = (
          <fbt desc="Lists">
            Available Locations: <fbt:list name="locations" delimiter="bullet" items={["Tokyo", "London", "Vienna"]} />.
          </fbt>
        );`,
    ),
  ).toMatchSnapshot();

  expect(
    transform(
      `const x = (
          <fbt desc="Lists">
            Available Locations: <fbt:list name="locations" conjunction="or" delimiter="bullet" items={["Tokyo", "London", "Vienna"]} />.
          </fbt>
        );`,
    ),
  ).toMatchSnapshot();
});
