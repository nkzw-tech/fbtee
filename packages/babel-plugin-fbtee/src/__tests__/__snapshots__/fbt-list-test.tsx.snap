// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<fbt:list> 1`] = `
import { fbt } from "fbtee";
const x = fbt._(
  "Available Locations: {locations}.",
  [fbt._list("locations", ["Tokyo", "London", "Vienna"])],
  {
    hk: "1b9IBP",
  }
);

`;

exports[`<fbt:list> 2`] = `
import { fbt } from "fbtee";
const x = fbt._(
  "Available Locations: {locations}.",
  [fbt._list("locations", ["Tokyo", "London", "Vienna"], "and")],
  {
    hk: "1b9IBP",
  }
);

`;

exports[`<fbt:list> 3`] = `
import { fbt } from "fbtee";
const x = fbt._(
  "Available Locations: {locations}.",
  [fbt._list("locations", ["Tokyo", "London", "Vienna"], null, "bullet")],
  {
    hk: "1b9IBP",
  }
);

`;

exports[`<fbt:list> 4`] = `
import { fbt } from "fbtee";
const x = fbt._(
  "Available Locations: {locations}.",
  [fbt._list("locations", ["Tokyo", "London", "Vienna"], "or", "bullet")],
  {
    hk: "1b9IBP",
  }
);

`;

exports[`fbt.list() 1`] = `
import { fbt } from "fbtee";
fbt._(
  "Available Locations: {locations}",
  [fbt._list("locations", ["Tokyo", "London", "Vienna"], null, "or")],
  {
    hk: "19372u",
  }
);

`;
